pymodbus/client/sync.py

```diff
@@ -256,9 +261,9 @@ def _recv(self, size):
 
         timeout = self.timeout
 
-        # If size isn't specified read 1 byte at a time.
+        # If size isn't specified read up to 4096 bytes at a time.
         if size is None:
-            recv_size = 1
+            recv_size = 4096
```

Comment:
This value is debatable. This would essentially block the `recv` till all `4096` bytes are received or timeout. Adding further to delay in reads.

Reply:
That's not how `recv` works. From the documentation:
```
$ pydoc socket.socket.recv | more
Help on method_descriptor in socket.socket:

socket.socket.recv = recv(...)
    recv(buffersize[, flags]) -> data
    
    Receive up to buffersize bytes from the socket.  For the optional flags
    argument, see the Unix manual.  When no data is available, block until
    at least one byte is available or until the remote end is closed.  When
    the remote end is closed and all data is read, return the empty string.
```

So it never blocks when a single byte is available, regardless of what `buffersize` is provided. I.e. it's a maximum limit, the minimum limit is always 1 regardless.

Reply:
@thormick That matches the underlying recv man page too:

https://man7.org/linux/man-pages/man2/recv.2.html
> If no messages are available at the socket, the receive calls wait for a message to arrive, unless the socket is nonblocking (see fcntl(2)), in which case the value -1 is returned and the external variable errno is set to EAGAIN or EWOULDBLOCK.  The receive calls normally return any data available, up to the requested amount, rather than waiting for receipt of the full amount requested.

Reply:
ah, I got this confused with `Serial`.Apologies for the mistake. Looks good.

Reply:
Actually, it's good that you picked on it, it made me gave it a quick look over and I realize now that `ModbusTlsClient` does the same, and now those two will be inconsistent. I don't have a setup ready for testing modbus TLS, so I'm not sure about touching that part of the code. Not necessarily a problem, though?